src/content/structured/get-started/migrating-to-v3.mdx

```diff
@@ -0,0 +1,149 @@
+    rel="noreferrer noopener"
+  >
+    changelog
+  </ic-link>
+  .
+</p>
+
+## Theming and dark mode
+
+With V3's arrival comes an improved `ic-theme` component. This component was initially responsible for managing the theme of the UI Kit. In addition to the existing functionality, you can now use the `ic-theme` component to set whether a component responds to changes in the system's light and dark theme or whether it always appears in a particular mode. Guidance on implementing the `ic-theme` package can be found in the [custom theme guidance](/get-started/development/custom-theme).
+
+## Codemod
+
+To aid your migration from ICDS V2 to V3, we have created a codemod to automate the process of migrating your code. A codemod is a tool that can be used to make large-scale changes to codebases quickly and efficiently. Our ICDS V2 to V3 codemod is the simplest and easiest way to migrate your codebase to the latest version of the UI Kit.
+
+### How it works
+
+Given a directory, it will scan over files and find any relevant ICDS components that require a change either by component name or props used.
+
+### How to run
+
+<p>
+  This package works as an executable requiring a directory and optional test
+  boolean argument to help migrate your tests from ICDS V2 to ICDS V3. Further
+  information and examples can be found in the{" "}
+  <ic-link
+    target="_blank"
+    href="https://github.com/mi6/ic-ui-kit/blob/v3.0.0/main/packages/codemod/README.md"
+    rel="noreferrer noopener"
+  >
+    codemod package README
+  </ic-link>
+  .
+</p>
+
+```console
+- npm @ukic/react@latest
+- npm @ukic/web-components@latest
+- npx @ukic/codemod <dir> <test>
+```
+
+## Breaking changes
+
+### 1. Prop renaming
+
+- The `appearance` prop has been replaced with the `theme` and `monochrome` [props](/get-started/development/custom-theme/#dark-mode-as-a-theme).
+- The `auto-focus` prop has been renamed to `autofocus`.
+- The `color` prop has been renamed to `brandColor`.
+- The `group-title` prop in `ic-accordion-group` (`IcAccordionGroup`) and `ic-footer-link-group` (`IcFooterLinkGroup`) has been renamed to `label`.
+- The `text-label` prop in `ic-badge` (`IcBadge`) has been renamed to `label`.
+- The `body-max-lines` prop in `ic-empty-state` (`IcEmptyState`) has been renamed to `max-lines`.
+- The `error` prop in `ic-input-label` (`IcInputLabel`) has been renamed to `status` with `error` as an option.
+- The `boundary-count` and `adjacent-count` props in `ic-pagination` (`IcPagination`) have been renamed to `boundary-page-count` and `adjacent-page-count`.
+- The `disable-filter` prop in `ic-search-bar` (`IcSearchBar`) and `ic-select` (`IcSelect`) has been renamed to `disable-auto-filtering`.
+- The `hint-text` prop in `ic-search-bar` (`IcSearchBar`) has been renamed to `assistive-hint-text`.
+- Some `ic-step` (`IcStep`) props have had the prefix "step" removed, e.g., `step-status` has been renamed to `status`.
```

Comment:
Please add all the prop changes in IcStep as well as any missing props mentioned in the [other migration PR](https://github.com/mi6/ic-design-system/pull/1264/files#diff-064a856c0f72ab51a0d8801a34b9bfdfd414915d9c81976ff2cc1beede7f4aba). I've seen that the `keyboardShortcut` prop rename in `ic-menu-item` is missing but haven't checked if any others are missing